### PR TITLE
API Impl: Checkout and User Order Transaction History

### PR DESCRIPTION
**Pull request by brentmillen**
## Ticket/s

- [Checkout Impl](https://trello.com/c/Y0Tce43D/115-checkout-api-implementation)
- [Checkout Page](https://trello.com/c/UxFEhBZP/145-checkout-page)

### Major Id related changes:
- Changed stored and referenced userId to the objectId of user schema for the following modules(previously authId):
  - Cart
  - Order 
  - Merchant
  - Products
### - Created Checkout Page 
- Protect routes for user with cart
### - Created User Transaction View
- Protect routes for user only

## Screenshot

image
image

### Make sure to create new records for the entire process to apply updates(new merchants, products, user etc.)
- Navigate to /cart (make sure you already have items to checkout)
- Click checkout
- Click place order
- See orders
- Change status in DB into delivered to see contents in delivered tab, as well as rate button

### Additional testing:
- Navigate to /checkout and /order as non-user or merchant
- Navigate to /checkout without cart items

### Note: This integration can be polished in another PR, needed to merge ASAP so we can start with Rating Impl

### CODE REVIEW

**Comment by brentmillen** [comment]
found bug: create order API has no exception handler if items checked out are not in cart. Although it won't create an order document as expected but the response is an empty array and status is 201(created).

File: packages/ui/transaction-history/Order.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                    Status
+                  </Text>
+                  <Divider orientation="vertical" />
+                  <Text
+                    fz="sm"
+                    fw={700}
+                    color="#FFC815"
+                    style={{ display: 'inline-block' }}
+                  >
+                    To Ship
```
Comment (TreshAnn) [inline packages/ui/transaction-history/Order.tsx]:
```suggestion
                    {data.status}
```

File: packages/ui/transaction-history/Order.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                  </Text>
+                </Group>
+              </td>
+              <td colSpan={3}>
```
Comment (TreshAnn) [inline packages/ui/transaction-history/Order.tsx:128]:
```suggestion
              <td colSpan={4}>
```

**Review by TreshAnn — APPROVED** [review]
I've reviewed your changes, LGTM! Please take a look at the comments for some minor suggestions. Once those are addressed, feel free to merge this PR. thanks.

File: packages/ui/transaction-history/Order.tsx (hunk 2 above)
Comment (brentmillen) [reply packages/ui/transaction-history/Order.tsx:128]: done applied in first td

**Review by sandrex-SGV — APPROVED** [review]
LGTM!

**Pull request merged**